File: I-DataStructures/2-Linked-List/linked-list.go
package main

import (
	"fmt"
)

type Node struct{
	data interface{}
	next *Node
}

type LinkedList struct{
	head *Node
}

func (list *LinkedList) Insert(d interface{}) {
	newNode := &Node{data: d, next: nil}
	if list.head == nil{
		list.head = newNode
	} else {
		pointer := list.head
		for pointer.next != nil {
			pointer = pointer.next
		}
		pointer.next = newNode
	}
}

func (list *LinkedList) Show(){
	pointer := list.head
	for pointer != nil {
		fmt.Printf("%v -> ", pointer.data)
		pointer = pointer.next
	}
	fmt.Print("NULL\n")
}

func main(){
	fmt.Println("Linked Lists in Go")
	// Init the list
	list := LinkedList{}
	list.Insert(22)
	list.Insert(30)
	list.Insert(57)
	list.Show()
}

File: I-DataStructures/2-Linked-List/linked-list.py
class Node:
    def __init__(self, data=None):
        self.data = data
        self.next = None
     
     
class LinkedList:
    def __init__(self):
        self.head = None
        
    def insert(self, data=None):
        new_node = Node(data)
        if self.head is None:
            self.head = new_node
        else:
            pointer = self.head
            while pointer.next is not None:
                pointer = pointer.next
            pointer.next = new_node
            
    def show(self):
        pointer = self.head
        while pointer is not None:
            print(pointer.data, "-> ", end='')
            pointer = pointer.next
        print("NULL")


def main():
    print("Linked Lists in Python")
    linkedlist = LinkedList()
    linkedlist.insert(data=22)
    linkedlist.insert(data=30)
    linkedlist.insert(data=57)
    linkedlist.show()
        
        
if __name__ == "__main__":
    main()
